### PR TITLE
Add PySUS library to insert data from SINAN

### PR DESCRIPTION
**Pull request by esloch**
## Description

following [PR134](https://github.com/thegraphnetwork/EpiGraphHub/pull/134).

## Related Issue

[Issue 123](https://github.com/thegraphnetwork/EpiGraphHub/issues/123) from EpiGraphHub.

## Checklist

- [x] I've read the [`CODE_OF_CONDUCT.md`](https://github.com/thegraphnetwork/epigraphhub_py/blob/master/CODE_OF_CONDUCT.md) document.
- [x] I've read the [`CONTRIBUTING.md`](https://github.com/thegraphnetwork/epigraphhub_py/blob/master/CONTRIBUTING.md) guide.
- [x] I've used pre-commit hook locally.
- [x] I've written tests for all new methods and classes that I created.
- [x] I've written the docstring in Google format for all the methods and classes that I used.
- [x] Semantic-release prefix title used:
  - Use the prefix `chore(docs):` for Examples / docs / tutorials / dependencies update
  - Use the prefix `fix:` for a bug fix
    (non-breaking change which fixes an issue).
  - Use the prefix `chore(improvement):` for an improvement
    (non-breaking change which improves an existing feature).
  - Use the prefix `feat:` for a new feature
    (non-breaking change which adds functionality).
  - Use the prefix `breaking change:` for a breaking change
    (fix or feature that would cause existing functionality to change).
  - Use the prefix `chore(security):` for a security fix.

**Note**: The title prefix can be also expanded with more information
inside the parenthesis. Some examples for a full title (prefix + message):
- `fix(login): Fix the login page`.
- `feat(forecast): Add a weather forecast page`.
- `chore(security-login): Add a weather forecast page`.

### CODE REVIEW

File: epigraphhub/data/sinan_fetch.py [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+PSQL_URI = "postgresql://{}:{}@{}:{}/{}".format(
+    db_config.get("user"),
+    db_config.get("password"),
+    db_config.get("host"),
+    db_config.get("port"),
+    db_config.get("database"),
+)
+
+engine = create_engine(PSQL_URI)
+conn = psycopg2.connect(**db_config)
```
Comment (eduardocorrearaujo) [inline epigraphhub/data/sinan_fetch.py:38]: We have the functions that Ivan created in this script to make the connection: https://github.com/thegraphnetwork/epigraphhub_py/blob/6f76dc302f7191e6c86c78196cc18640d3cf5b32/epigraphhub/settings.py#L1

example of use: 

```
from sqlalchemy import create_engine
from epigraphhub.settings import env

with env.db.credentials[env.db.default_credential] as credential:
    engine_public = create_engine(
        f"postgresql://{credential.username}:"
        f"{credential.password}@{credential.host}:{credential.port}/"
        f"{credential.dbname}"
    )
```

It's just necessary to config the credentials

File: tests/test_data/test_sinan_fetch.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        parquet_to_df = schema_from_df(self.fname)[2]
+
+        self.assertEqual(table_name, '"brasil"."zikabr17"')
+        self.assertIn("CREATE TABLE brasil.zikabr17", query_psql)
+        self.assertIsInstance(parquet_to_df, pd.DataFrame)
+        self.assertTrue(self.fname_pq.exists())
+
+    def test_parquet_to_df(self):
+        df_compare = parquet_to_df(self.fname)
+        self.assertEqual(df_compare.shape[0], 32684)
```
Comment (eduardocorrearaujo) [inline tests/test_data/test_sinan_fetch.py:34]: The size of this file can change, right? What could break the test since the number of rows is hard coded

Comment (esloch) [reply tests/test_data/test_sinan_fetch.py:34]: That's right, the dataframe size will never change. The same parquet will always be downloaded: *self.fname = "ZIKABR17.parquet*, for this case if the result is not the same as **32684**, something was not successful in concatenating the dataframes.

Comment (eduardocorrearaujo) [reply tests/test_data/test_sinan_fetch.py:34]: Ok. I was afraid that the dataframe size could change with some later updates.

Comment (esloch) [reply tests/test_data/test_sinan_fetch.py:34]: > Ok. I was afraid that the dataframe size could change with some later updates.

If I'm not mistaken, the *PRELIMINAR* datasus data may change, but these are datasets after 2019.

File: epigraphhub/data/sinan_fetch.py [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+            use_legacy_dataset=False,
+        )
+        .read_pandas()  # columns=COL_NAMES
+        .to_pandas()
+    )
+    
+    # Measure execution time ended
+    elapsed_time = time.time() - st
+    print(
+        "Get data from PySUS to {}, finished in: {}".format(
```
Comment (fccoelho) [inline epigraphhub/data/sinan_fetch.py:60]: replace PYSUS by SUS. PySUS is a library not a place.

**Review by fccoelho — APPROVED** [review]

File: epigraphhub/data/sinan_fetch.py (hunk 1 above)
Comment (esloch) [reply epigraphhub/data/sinan_fetch.py]: Thank you!

**Comment by fccoelho** [comment]
Latest CI Build is failing because of the issue of PyMC3 dependncy that is already fixed in PR #180

File: tests/test_data/test_sinan_fetch.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        df1 = pd.read_sql(query, engine)
+        df2 = parquet_to_df(self.fname)
+        query_sql = get_schema(df2, self.table, schema=self.schema, con=engine)
+
+        self.assertEqual(table_name, '"brasil"."zikabr17"')
+        self.assertIn("CREATE TABLE brasil.zikabr17", query_sql)
+        self.assertEqual(int(df1["count"]), df2.shape[0])
+
+
+if __name__ == "__main__":
```
Comment (xmnlab) [inline tests/test_data/test_sinan_fetch.py:50]: probably you can remove this line, as we are using pytest

File: epigraphhub/data/sinan_fetch.py [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+import psycopg2.extras as extras
+import pyarrow.parquet as pq
+from loguru import logger
+from pandas.io.sql import get_schema
+from pysus.online_data import SINAN
+from sqlalchemy import create_engine
+
+from epigraphhub.settings import env
+
+"""
```
Comment (xmnlab) [inline epigraphhub/data/sinan_fetch.py:16]: you can remove this comments

File: epigraphhub/data/sinan_fetch.py [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+                    df.shape[1],
+                    time.strftime("%H:%M:%S", time.gmtime(elapsed_time)),
+                )
+            )
+
+        except Exception as e:
+            logger.info(e)
+
+
+"""
```
Comment (xmnlab) [inline epigraphhub/data/sinan_fetch.py:176]: you can remove this commented code

**Comment by xmnlab** [comment]
LGTM! thanks @esloch for working on that!

**Comment by esloch** [comment]
Thanks everyone for the review!

**Pull request merged**